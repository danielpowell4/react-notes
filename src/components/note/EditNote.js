import React from 'react';
import {partial} from '../../lib/utils';

export const EditNote = (props) => {
  const handleUpdateContent = partial(props.handleUpdateContent, props.id)
  const handleEditSubmit = partial(props.handleEditSubmit, props.id)
  const content = props.content

  return (<form onSubmit={handleEditSubmit}>
    <input type="text"
      onChange={handleUpdateContent}
      value={content}/>
  </form>)
}

  EditNote.propTypes = {
    content: React.PropTypes.string.isRequired,
    handleEditSubmit: React.PropTypes.func.isRequired,
    handleUpdateContent: React.PropTypes.func.isRequired,
    id: React.PropTypes.number.isRequired
  }
