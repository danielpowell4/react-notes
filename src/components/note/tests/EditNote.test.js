import React from 'react';
import renderer from 'react-test-renderer';
import {EditNote} from '../EditNote';

/* EditNote snapshot test */

const editNoteSampleProps = { content: 'editted content',
                      handleEditSubmit: function(){},
                      handleUpdateContent: function(){},
                      id: 1 }

test('EditNote renders correctly', () => {
  const tree = renderer.create(
    <EditNote {...editNoteSampleProps} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
