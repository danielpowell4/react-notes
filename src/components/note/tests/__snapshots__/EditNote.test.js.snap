exports[`test EditNote renders correctly 1`] = `
<form
  onSubmit={[Function]}>
  <input
    onChange={[Function]}
    type="text"
    value="editted content" />
</form>
`;
